### PR TITLE
Add Open Fodder

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/OpenFodder/openfodder/issues/58
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://github.com/OpenFodder/openfodder/pull/59

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build org.openfodder.OpenFodder

File: openfodder.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>openfodder.desktop</id>
```
Comment (hfiguiere) [inline openfodder.appdata.xml:3]: you should put the right ID and name the file with the right name.

File: openfodder.desktop [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
```
Comment (hfiguiere) [inline openfodder.desktop:1]: same about the name.

File: openfodder.desktop [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+Name=Open Fodder
+GenericName=Action-Strategy Shoot 'em 
+Comment=An open source port of Cannon Fodder
+Icon=openfodder
```
Comment (hfiguiere) [inline openfodder.desktop:5]:
```suggestion
Icon=org.openfodder.OpenFodder
```

File: org.openfodder.OpenFodder.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
+app-id: org.openfodder.OpenFodder
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
```
Comment (hfiguiere) [inline org.openfodder.OpenFodder.yml:3]:
```suggestion
runtime-version: '21.08'
```

consider 20.08 deprecated for new packages.

File: org.openfodder.OpenFodder.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        path: openfodder.png
+      - type: file
+        path: openfodder.appdata.xml
+    build-commands:
+      - make -j${FLATPAK_BUILDER_N_JOBS}
+    post-install:
+      - mkdir -p ${FLATPAK_DEST}/share/OpenFodder/
+      - cp -r  Run/* ${FLATPAK_DEST}/share/OpenFodder/
+      - install -Dm755 openfodder.sh ${FLATPAK_DEST}/bin/openfodder
+      - install -Dm644 openfodder.appdata.xml -t ${FLATPAK_DEST}/share/metainfo/
```
Comment (hfiguiere) [inline org.openfodder.OpenFodder.yml:47]: as mentionned above, you might as well have all the names right from the get go.

File: org.openfodder.OpenFodder.yml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  - org.freedesktop.Sdk.Extension.llvm11
+build-options:
+  append-path: /usr/lib/sdk/llvm11/bin
+  prepend-ld-library-path: /usr/lib/sdk/llvm11/lib
+command: openfodder
+rename-appdata-file: openfodder.appdata.xml
+rename-desktop-file: openfodder.desktop
+rename-icon: openfodder
+finish-args:
+  - --socket=wayland
```
Comment (hfiguiere) [inline org.openfodder.OpenFodder.yml:15]: does it really support wayland?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.openfodder.OpenFodder.yml (hunk 4 above)
Comment (Mailaender) [reply org.openfodder.OpenFodder.yml:3]: I rely on an older version of LLVM.

File: org.openfodder.OpenFodder.yml (hunk 6 above)
Comment (Mailaender) [reply org.openfodder.OpenFodder.yml:15]: I don't know. Can't test as Wayland is effectively broken on my machines. It uses SDL2.

**Comment by Mailaender** [comment]
bot, build org.openfodder.OpenFodder

**Comment by Mailaender** [comment]
bot, build org.openfodder.OpenFodder

File: org.openfodder.OpenFodder.yml (hunk 4 above)
Comment (hfiguiere) [reply org.openfodder.OpenFodder.yml:3]: it doesn't build with llvm12 or 13?

Not sure why llvm11 doesn't come for a more recent runtime, but there is not point in going that route.

File: org.openfodder.OpenFodder.yml (hunk 6 above)
Comment (Mailaender) [reply org.openfodder.OpenFodder.yml:12]: The controls are also a bit wonky on X11.

File: org.openfodder.OpenFodder.yml (hunk 4 above)
Comment (hfiguiere) [reply org.openfodder.OpenFodder.yml:3]: just to clarify. 20.08 runtime will be EOL in 22.08. 

you should build this with a more recent compiler. Chances that it doesn't support a more recent compiler is a bug.

Comment (Mailaender) [reply org.openfodder.OpenFodder.yml:3]: 12 and 13 spit out error messages.

File: org.openfodder.OpenFodder.yml (hunk 6 above)
Comment (Mailaender) [reply org.openfodder.OpenFodder.yml]: Dropped Wayland.

**Comment by nedrichards** [comment]
bot, build org.openfodder.OpenFodder

**Comment by barthalion** [comment]
bot, build org.openfodder.OpenFodder

**Comment by barthalion** [comment]
bot, build org.openfodder.OpenFodder

**Pull request closed without merging**